src/main/java/org/candlepin/controller/refresher/RefreshWorker.java

```diff
@@ -508,18 +578,43 @@ public RefreshResult execute(Owner owner) {
             this.contentMapper.clearExistingEntities();
 
             // Add in our existing entities
-            this.poolMapper.addExistingEntities(
-                this.poolCurator.listByOwnerAndTypes(owner.getId(), PoolType.NORMAL, PoolType.DEVELOPMENT));
+            List<Pool> pools = this.poolCurator
```

Comment:
Do we need obtain a system lock for the poolCurator?

Reply:
Probably not. At least, not at present. Pools are currently gated behind the owner locks elsewhere in refresh and the various pool manipulating operations. If pool scope changes, or we give our locking model a proper pass/refactor, that'd definitely be something to consider.